transmit/direct_transmit.go

```diff
@@ -334,7 +364,20 @@ func (d *DirectTransmission) sendBatch(wholeBatch []*types.Event) {
 			d.handleBatchFailure(subBatch)
 			continue
 		}
-		req, err := http.NewRequest("POST", apiURL.String(), bytes.NewReader(packed))
+
+		var reader io.Reader
+		if d.enableCompression {
+			zBufPtr := batchBufferPool.Get().(*[]byte)
+			zBuf := zstdEncoder.EncodeAll(packed, *zBufPtr)
+			reader = bytes.NewReader(zBuf)
+
+			*zBufPtr = zBuf[:0]
+			defer batchBufferPool.Put(zBufPtr)
```

Comment:
Should we put it back per batch instead of the whole function is finished?

Reply:
It doesn't really matter as long as it goes back eventually, and given the various error handling below, this is simpler.